---
title: "Oral Microbiome - Most abundant across site `r Sys.Date()`"
author: "Florentin CONSTANCIAS"
date: "`r Sys.Date()`"
output: 
  html_document:
    toc: true
    toc_depth: 4
---

<style type="text/css">
div.main-container {
max-width: 1800px;
margin-left: auto;
margin-right: auto;
}
.Table-Normal {
position: relative;
//display: block;
margin: 10px auto;
padding: 0;
width: 100%;
height: auto;
border-collapse: collapse;
text-align: center;
}
</style>


```{r setup, include=FALSE}
rm(list = ls())

# knitr::opts_knit$set(root.dir = "/Users/fconstan/Projects/Oral/") 
knitr::opts_chunk$set(echo = TRUE)
DT::datatable(matrix())
# knitr::opts_chunk$set(dev = 'pdf')
# knitr::purl("Mdown.Rmd")

```


```{r packages, results=TRUE, warning=TRUE, include=FALSE}
require(tidyverse); packageVersion("tidyverse")
require(phyloseq); packageVersion("phyloseq")
```

# Fin most abundant taxa across sites:

```{r, message = FALSE, warning = FALSE, include=FALSE}
biobakery_function <- ("~/Documents/GitHub/DivComAnalyses/R/phyloseq_biobakery_functions.R")
alpha <- ("~/Documents/GitHub/DivComAnalyses/R/phyloseq_alpha.R")
beta <- ("~/Documents/GitHub/DivComAnalyses/R/phyloseq_beta.R")
taxa <- ("~/Documents/GitHub/DivComAnalyses/R/phyloseq_taxa_tests.R")
norm <- ("~/Documents/GitHub/DivComAnalyses/R/phyloseq_normalisation.R")
heat <- ("~/Documents/GitHub/DivComAnalyses/R/phyloseq_heatmap.R")
varia <- ("~/Documents/GitHub/DivComAnalyses/R/phyloseq_varia.R")

```

```{r, message = FALSE, results= "hide", warning = FALSE, include = TRUE}
sapply(c(biobakery_function, alpha, beta, taxa, norm, heat, varia), source, chdir = TRUE)
```


# Data loading:

```{r, message = FALSE, results=FALSE}
input <- list(metpahlan_ps = "~/Documents/GitHub/oral-microbiome/data/processed/metaphlan_ps.RDS",
              motus_ps = "~/Documents/GitHub/oral-microbiome/data/processed/motus/physeq.RDS",
              pathway_ps = "~/Documents/GitHub/oral-microbiome/data/processed/humann_pw_ps.RDS",
              kegg_ps = "~/Documents/GitHub/oral-microbiome/data/processed/humann_kegg_ps.RDS",
              l4c_ps = "~/Documents/GitHub/oral-microbiome/data/processed/humann_l4c_ps.RDS",
              go_ps = "~/Documents/GitHub/oral-microbiome/data/processed/humann_go_ps.RDS",
              infogo_ps = "~/Documents/GitHub/oral-microbiome/data/processed/humann_infogo_ps.RDS",
              metac_ps = "~/Documents/GitHub/oral-microbiome/data/processed/humann_metac_ps.RDS",
              pfam_ps = "~/Documents/GitHub/oral-microbiome/data/processed/humann_pfam_ps.RDS")

```


```{r, warning =FALSE, message = FALSE}
purrr::map(input, 
           readRDS) -> objects
```


```{r, message = FALSE, warning = FALSE}
sample_data(objects$metpahlan_ps$physeq)$Site_Health <- paste0(sample_data(objects$metpahlan_ps$physeq)$Oral_Site,
                                                               "_",
                                                               sample_data(objects$metpahlan_ps$physeq)$Health_status)
```



Load gene abundance table and have a look:

```{r message=FALSE, warnings = FALSE}
objects$kegg_ps$physeq  %>%
  humann2_species_contribution(meta_data_var = c("Subject", "sample_Sample" ,"Type", "Oral_Site", "Health_status")) %>%
  separate(Feature, into = c("code", "name"), sep = ": ", remove = FALSE) %>%
  dplyr::filter(!is.na(name))  -> gn

gn %>%
  select(-id, -code) %>%
  head(n = 1000) %>%
  distinct(name, .keep_all = TRUE) %>%
  DT::datatable()
```

```{r fig.height = 6 , fig.width = 16 , message=FALSE}
pwy %>%
  filter(Species %in% strep_sel) %>%
  select(-id, -sample_Sample, Health_status) %>%
  group_by(Subject, Oral_Site) %>%
  add_count(name = "count_per") %>% 
  slice_max(order_by = DNA, n = 5) %>%
DT::datatable()
```
```{r fig.height = 6 , fig.width = 16 , message=FALSE}
pwy %>%
  filter(Species %in% strep_sel) %>%
  select(-id, -sample_Sample, Health_status) %>%
  group_by(Subject, Oral_Site) %>%
  add_count(name = "count_per") %>% 
  slice_max(order_by = RNA, n = 5) %>%
DT::datatable()
```

```{r fig.height = 6 , fig.width = 16 , message=FALSE}
pwy %>%
  filter(Species %in% strep_sel) %>%
  select(-id, -sample_Sample, Health_status) %>%
  group_by(Subject, Oral_Site) %>%
  add_count(name = "count_per") %>% 
  slice_max(order_by = RNA_DNA, n = 5) %>%
DT::datatable()
```
Fig 4. Gene activity of selected species


FigS2A: gene expression of selected streptococcus species in saliva vs. tongue in health
Fig4A: significant gene expression by candidate species saliva versus tongue in health

(cf S1A â€“ pathway expression of selected streptococcus species in saliva vs. tongue in health)

```{r fig.height = 2 , fig.width = 2 , message=FALSE}
pwy %>%
  dplyr::filter(Health_status == "Healthy control",
                Oral_Site %in% c("SALIVA", "TONGUE_BIOFILM")) %>%
  humann2_RNA_DNA_ratio_plot(x_plot = "name", 
                             y_plot = "log2(RNA_DNA)", 
                             color = "Oral_Site", 
                             fill = "Oral_Site",
                             shape = NULL,
                             filter_species = strep_sel,
                             facet_formula = " . ~ Health_status + Species",
                             export_legend = TRUE) -> plot

plot$legend
```

```{r fig.height = 40, fig.width = 22 , message=FALSE}
plot$plot
```

```{r , message=FALSE}
plot$plot$data %>% 
  # mutate(log2RNA_DNA = log2(RNA_DNA)) %>%
  ggpubr::compare_means(RNA_DNA ~  Oral_Site,
                        group.by = c("Species","Feature"),
                        data = .,
                        # method = "wilcox.test",
                        p.adjust.method = "fdr") %>%
  filter(p.adj < 0.05) -> diff_signif  

diff_signif %>%
  DT::datatable()
```

```{r fig.height = 20 , fig.width = 10 , message=FALSE}
pwy %>%
  dplyr::filter(Health_status == "Healthy control") %>%
     dplyr::filter(Oral_Site %in% c("SALIVA", "TONGUE_BIOFILM")) %>%
   # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
  humann2_RNA_DNA_ratio_plot(x_plot = "name", 
                       y_plot = "log2(RNA_DNA)", 
                       color = "Oral_Site", 
                       fill = "Oral_Site",
                       shape = NULL,
                       filter_species = "Streptococcus_infantis",
                       filter_feature = diff_signif %>% filter(Species == "Streptococcus_infantis") %>% pull(Feature),
                       facet_formula = " . ~ Health_status ",
                       export_legend = TRUE) -> plot
  

plot$legend
plot$plot
```


```{r fig.height = 20 , fig.width = 10 , message=FALSE}
pwy %>%
  dplyr::filter(Health_status == "Healthy control") %>%
     dplyr::filter(Oral_Site %in% c("SALIVA", "TONGUE_BIOFILM")) %>%
   # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
  humann2_RNA_DNA_ratio_plot(x_plot = "name", 
                       y_plot = "log2(RNA_DNA)", 
                       color = "Oral_Site", 
                       fill = "Oral_Site",
                       shape = NULL,
                       filter_species = "Streptococcus_parasanguinis",
                       filter_feature = diff_signif %>% filter(Species == "Streptococcus_parasanguinis") %>% pull(Feature),
                       facet_formula = " . ~ Health_status ",
                       export_legend = TRUE) -> plot
  

plot$legend
plot$plot
```


```{r fig.height = 20 , fig.width = 10 , message=FALSE}
pwy %>%
  dplyr::filter(Health_status == "Healthy control") %>%
     dplyr::filter(Oral_Site %in% c("SALIVA", "TONGUE_BIOFILM")) %>%
   # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
  humann2_RNA_DNA_ratio_plot(x_plot = "name", 
                       y_plot = "log2(RNA_DNA)", 
                       color = "Oral_Site", 
                       fill = "Oral_Site",
                       shape = NULL,
                       filter_species = "Streptococcus_salivarius",
                       filter_feature = diff_signif %>% filter(Species == "Streptococcus_salivarius") %>% pull(Feature),
                       facet_formula = " . ~ Health_status ",
                       export_legend = TRUE) -> plot
  

plot$legend
plot$plot
```



FigS2B: gene expression of selected streptococcus species in saliva vs. plaque in health
Fig4B: significant gene expression by candidate species saliva versus plaque in health

```{r fig.height = 2 , fig.width = 2 , message=FALSE}
pwy %>%
  dplyr::filter(Health_status == "Healthy control",
                Oral_Site %in% c("SALIVA", "SUBGINGIVAL_PLAQUE")) %>%
  humann2_RNA_DNA_ratio_plot(x_plot = "name", 
                             y_plot = "log2(RNA_DNA)", 
                             color = "Oral_Site", 
                             fill = "Oral_Site",
                             shape = NULL,
                             filter_species = strep_sel,
                             facet_formula = " . ~ Health_status + Species",
                             export_legend = TRUE) -> plot

plot$legend
```

```{r fig.height = 60 , fig.width = 22 , message=FALSE}
plot$plot
```

```{r , message=FALSE}
plot$plot$data %>% 
  # mutate(log2RNA_DNA = log2(RNA_DNA)) %>%
  ggpubr::compare_means(RNA_DNA ~  Oral_Site,
                        group.by = c("Species","Feature"),
                        data = .,
                        # method = "wilcox.test",
                        p.adjust.method = "fdr") %>%
  filter(p.adj < 0.05) -> diff_signif  

diff_signif %>%
  DT::datatable()
```

```{r fig.height = 20 , fig.width = 10 , message=FALSE}
# pwy %>%
#   dplyr::filter(Health_status == "Healthy control") %>%
#      dplyr::filter(Oral_Site %in% c("SALIVA", "SUBGINGIVAL_PLAQUE")) %>%
#    # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
#   humann2_RNA_DNA_ratio_plot(x_plot = "name", 
#                        y_plot = "log2(RNA_DNA)", 
#                        color = "Oral_Site", 
#                        fill = "Oral_Site",
#                        shape = NULL,
#                        filter_species = "Streptococcus_infantis",
#                        filter_feature = diff_signif %>% filter(Species == "Streptococcus_infantis") %>% pull(Feature),
#                        facet_formula = " . ~ Health_status ",
#                        export_legend = TRUE) -> plot
#   
# 
# plot$legend
# plot$plot
```


```{r fig.height = 20 , fig.width = 10 , message=FALSE}
# pwy %>%
#   dplyr::filter(Health_status == "Healthy control") %>%
#      dplyr::filter(Oral_Site %in% c("SALIVA", "SUBGINGIVAL_PLAQUE")) %>%
#    # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
#   humann2_RNA_DNA_ratio_plot(x_plot = "name", 
#                        y_plot = "log2(RNA_DNA)", 
#                        color = "Oral_Site", 
#                        fill = "Oral_Site",
#                        shape = NULL,
#                        filter_species = "Streptococcus_parasanguinis",
#                        filter_feature = diff_signif %>% filter(Species == "Streptococcus_parasanguinis") %>% pull(Feature),
#                        facet_formula = " . ~ Health_status ",
#                        export_legend = TRUE) -> plot
#   
# 
# plot$legend
# plot$plot
```


```{r fig.height = 20 , fig.width = 10 , message=FALSE}
# pwy %>%
#   dplyr::filter(Health_status == "Healthy control") %>%
#      dplyr::filter(Oral_Site %in% c("SALIVA", "SUBGINGIVAL_PLAQUE")) %>%
#    # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
#   humann2_RNA_DNA_ratio_plot(x_plot = "name", 
#                        y_plot = "log2(RNA_DNA)", 
#                        color = "Oral_Site", 
#                        fill = "Oral_Site",
#                        shape = NULL,
#                        filter_species = "Streptococcus_salivarius",
#                        filter_feature = diff_signif %>% filter(Species == "Streptococcus_salivarius") %>% pull(Feature),
#                        facet_formula = " . ~ Health_status ",
#                        export_legend = TRUE) -> plot
#   
# 
# plot$legend
# plot$plot
```



Fig 7. Gene activity of selected species
Fig7A-F: significant pathway expression by candidate species at each site in health and disease.

A - SALIVA


```{r fig.height = 60 , fig.width = 15 , message=FALSE}
pwy %>%
   dplyr::filter(Species %in% strep_sel) %>%
     dplyr::filter(Oral_Site %in% c("SALIVA")) %>%
   # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
  humann2_RNA_DNA_ratio_plot(x_plot = "name", 
                       y_plot = "log2(RNA_DNA)", 
                       color = "Health_status", 
                       fill = "Health_status",
                       shape = NULL,
                       # filter_species = "Streptococcus_infantis",
                       # filter_feature = diff_signif %>% filter(Species == "Streptococcus_infantis") %>% pull(Feature),
                       facet_formula = " . ~ Species ",
                       export_legend = TRUE) -> plot
```

```{r fig.height = 60 , fig.width = 15 , message=FALSE}

plot$legend
plot$plot
```

```{r , message=FALSE}
plot$plot$data %>% 
  # mutate(log2RNA_DNA = log2(RNA_DNA)) %>%
  ggpubr::compare_means(RNA_DNA ~  Health_status,
                        group.by = c("Species","Feature"),
                        data = .,
                        # method = "wilcox.test",
                        p.adjust.method = "fdr") %>%
  filter(p.adj < 0.05) -> diff_signif  

diff_signif %>%
  DT::datatable()
```


```{r fig.height = 10 , fig.width = 10 , message=FALSE}
pwy %>%
   dplyr::filter(Species %in% strep_sel) %>%
     dplyr::filter(Oral_Site %in% c("SALIVA")) %>%
   # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
  humann2_RNA_DNA_ratio_plot(x_plot = "name", 
                       y_plot = "log2(RNA_DNA)", 
                       color = "Health_status", 
                       fill = "Health_status",
                       shape = NULL,
                       # filter_species = "Streptococcus_infantis",
                       filter_feature = diff_signif %>% filter(Species == "Streptococcus_infantis") %>% pull(Feature),
                       facet_formula = " . ~ Species ",
                       export_legend = TRUE) -> plot
  

plot$legend
plot$plot
```

```{r fig.height = 20 , fig.width = 10 , message=FALSE}
pwy %>%
   dplyr::filter(Species %in% strep_sel) %>%
     dplyr::filter(Oral_Site %in% c("SALIVA")) %>%
   # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
  humann2_RNA_DNA_ratio_plot(x_plot = "name", 
                       y_plot = "log2(RNA_DNA)", 
                       color = "Health_status", 
                       fill = "Health_status",
                       shape = NULL,
                       # filter_species = "Streptococcus_infantis",
                       filter_feature = diff_signif %>% filter(Species == "Streptococcus_parasanguinis") %>% pull(Feature),
                       facet_formula = " . ~ Species ",
                       export_legend = TRUE) -> plot
  

plot$legend
plot$plot
```

```{r fig.height = 20 , fig.width = 10 , message=FALSE}
pwy %>%
   dplyr::filter(Species %in% strep_sel) %>%
     dplyr::filter(Oral_Site %in% c("SALIVA")) %>%
   # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
  humann2_RNA_DNA_ratio_plot(x_plot = "name", 
                       y_plot = "log2(RNA_DNA)", 
                       color = "Health_status", 
                       fill = "Health_status",
                       shape = NULL,
                       # filter_species = "Streptococcus_infantis",
                       filter_feature = diff_signif %>% filter(Species == "Streptococcus_salivarius") %>% pull(Feature),
                       facet_formula = " . ~ Species ",
                       export_legend = TRUE) -> plot
  

plot$legend
plot$plot
```



B - TONGUE_BIOFILM


```{r fig.height = 60 , fig.width = 15 , message=FALSE}
pwy %>%
   dplyr::filter(Species %in% strep_sel) %>%
     dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>%
   # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
  humann2_RNA_DNA_ratio_plot(x_plot = "name", 
                       y_plot = "log2(RNA_DNA)", 
                       color = "Health_status", 
                       fill = "Health_status",
                       shape = NULL,
                       # filter_species = "Streptococcus_infantis",
                       # filter_feature = diff_signif %>% filter(Species == "Streptococcus_infantis") %>% pull(Feature),
                       facet_formula = " . ~ Species ",
                       export_legend = TRUE) -> plot
```

```{r fig.height = 60 , fig.width = 15 , message=FALSE}

plot$legend
plot$plot
```

```{r , message=FALSE}
plot$plot$data %>% 
  # mutate(log2RNA_DNA = log2(RNA_DNA)) %>%
  ggpubr::compare_means(RNA_DNA ~  Health_status,
                        group.by = c("Species","Feature"),
                        data = .,
                        # method = "wilcox.test",
                        p.adjust.method = "fdr") %>%
  filter(p.adj < 0.05) -> diff_signif  

diff_signif %>%
  DT::datatable()
```
```{r , message=FALSE}
plot$plot$data %>% 
  # mutate(log2RNA_DNA = log2(RNA_DNA)) %>%
  ggpubr::compare_means(RNA_DNA ~  Health_status,
                        group.by = c("Species","Feature"),
                        data = .,
                        # method = "wilcox.test",
                        p.adjust.method = "fdr") %>%
  filter(p < 0.05) -> diff_signif  

diff_signif %>%
  DT::datatable()
```


```{r fig.height = 10 , fig.width = 10 , message=FALSE}
pwy %>%
   dplyr::filter(Species %in% strep_sel) %>%
     dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>%
   # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
  humann2_RNA_DNA_ratio_plot(x_plot = "name", 
                       y_plot = "log2(RNA_DNA)", 
                       color = "Health_status", 
                       fill = "Health_status",
                       shape = NULL,
                       # filter_species = "Streptococcus_infantis",
                       filter_feature = diff_signif %>% filter(Species == "Streptococcus_infantis") %>% pull(Feature),
                       facet_formula = " . ~ Species ",
                       export_legend = TRUE) -> plot
  

plot$legend
plot$plot
```

```{r fig.height = 20 , fig.width = 10 , message=FALSE}
pwy %>%
   dplyr::filter(Species %in% strep_sel) %>%
     dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>%
   # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
  humann2_RNA_DNA_ratio_plot(x_plot = "name", 
                       y_plot = "log2(RNA_DNA)", 
                       color = "Health_status", 
                       fill = "Health_status",
                       shape = NULL,
                       # filter_species = "Streptococcus_infantis",
                       filter_feature = diff_signif %>% filter(Species == "Streptococcus_parasanguinis") %>% pull(Feature),
                       facet_formula = " . ~ Species ",
                       export_legend = TRUE) -> plot
  

plot$legend
plot$plot
```

```{r fig.height = 20 , fig.width = 10 , message=FALSE}
pwy %>%
   dplyr::filter(Species %in% strep_sel) %>%
     dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>%
   # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
  humann2_RNA_DNA_ratio_plot(x_plot = "name", 
                       y_plot = "log2(RNA_DNA)", 
                       color = "Health_status", 
                       fill = "Health_status",
                       shape = NULL,
                       # filter_species = "Streptococcus_infantis",
                       filter_feature = diff_signif %>% filter(Species == "Streptococcus_salivarius") %>% pull(Feature),
                       facet_formula = " . ~ Species ",
                       export_legend = TRUE) -> plot
  

plot$legend
plot$plot
```


C - SUBGINGIVAL_PLAQUE


```{r fig.height = 60 , fig.width = 15 , message=FALSE}
pwy %>%
   dplyr::filter(Species %in% strep_sel) %>%
     dplyr::filter(Oral_Site %in% c("SUBGINGIVAL_PLAQUE")) %>%
   # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
  humann2_RNA_DNA_ratio_plot(x_plot = "name", 
                       y_plot = "log2(RNA_DNA)", 
                       color = "Health_status", 
                       fill = "Health_status",
                       shape = NULL,
                       # filter_species = "Streptococcus_infantis",
                       # filter_feature = diff_signif %>% filter(Species == "Streptococcus_infantis") %>% pull(Feature),
                       facet_formula = " . ~ Species ",
                       export_legend = TRUE) -> plot
```

```{r fig.height = 60 , fig.width = 15 , message=FALSE}

plot$legend
plot$plot
```

```{r , message=FALSE}
plot$plot$data %>% 
  # mutate(log2RNA_DNA = log2(RNA_DNA)) %>%
  ggpubr::compare_means(RNA_DNA ~  Health_status,
                        group.by = c("Species","Feature"),
                        data = .,
                        # method = "wilcox.test",
                        p.adjust.method = "fdr") %>%
  filter(p.adj < 0.05) -> diff_signif  

diff_signif %>%
  DT::datatable()
```

```{r fig.height = 10 , fig.width = 10 , message=FALSE}
# pwy %>%
#    dplyr::filter(Species %in% strep_sel) %>%
#      dplyr::filter(Oral_Site %in% c("SUBGINGIVAL_PLAQUE")) %>%
#    # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
#   humann2_RNA_DNA_ratio_plot(x_plot = "name", 
#                        y_plot = "log2(RNA_DNA)", 
#                        color = "Health_status", 
#                        fill = "Health_status",
#                        shape = NULL,
#                        # filter_species = "Streptococcus_infantis",
#                        filter_feature = diff_signif %>% filter(Species == "Streptococcus_infantis") %>% pull(Feature),
#                        facet_formula = " . ~ Species ",
#                        export_legend = TRUE) -> plot
#   
# 
# plot$legend
# plot$plot
```

```{r fig.height = 20 , fig.width = 10 , message=FALSE}
# pwy %>%
#    dplyr::filter(Species %in% strep_sel) %>%
#      dplyr::filter(Oral_Site %in% c("SUBGINGIVAL_PLAQUE")) %>%
#    # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
#   humann2_RNA_DNA_ratio_plot(x_plot = "name", 
#                        y_plot = "log2(RNA_DNA)", 
#                        color = "Health_status", 
#                        fill = "Health_status",
#                        shape = NULL,
#                        # filter_species = "Streptococcus_infantis",
#                        filter_feature = diff_signif %>% filter(Species == "Streptococcus_parasanguinis") %>% pull(Feature),
#                        facet_formula = " . ~ Species ",
#                        export_legend = TRUE) -> plot
#   
# 
# plot$legend
# plot$plot
```

```{r fig.height = 20 , fig.width = 10 , message=FALSE}
# pwy %>%
#    dplyr::filter(Species %in% strep_sel) %>%
#      dplyr::filter(Oral_Site %in% c("SUBGINGIVAL_PLAQUE")) %>%
#    # dplyr::filter(Oral_Site %in% c("TONGUE_BIOFILM")) %>% 
#   humann2_RNA_DNA_ratio_plot(x_plot = "name", 
#                        y_plot = "log2(RNA_DNA)", 
#                        color = "Health_status", 
#                        fill = "Health_status",
#                        shape = NULL,
#                        # filter_species = "Streptococcus_infantis",
#                        filter_feature = diff_signif %>% filter(Species == "Streptococcus_salivarius") %>% pull(Feature),
#                        facet_formula = " . ~ Species ",
#                        export_legend = TRUE) -> plot
#   
# 
# plot$legend
# plot$plot
```



```{r , message=FALSE}
sessionInfo()
```









