---
title: "Oral Microbiome - Most abundant across site `r Sys.Date()`"
author: "Florentin CONSTANCIAS"
date: "`r Sys.Date()`"
output: 
  html_document:
    toc: true
    toc_depth: 4
---

<style type="text/css">
div.main-container {
max-width: 1800px;
margin-left: auto;
margin-right: auto;
}
.Table-Normal {
position: relative;
//display: block;
margin: 10px auto;
padding: 0;
width: 100%;
height: auto;
border-collapse: collapse;
text-align: center;
}
</style>


```{r setup, include=FALSE}
rm(list = ls())

# knitr::opts_knit$set(root.dir = "/Users/fconstan/Projects/Oral/") 
knitr::opts_chunk$set(echo = TRUE)
DT::datatable(matrix())
# knitr::opts_chunk$set(dev = 'pdf')
# knitr::purl("Mdown.Rmd")

```


```{r packages, results=TRUE, warning=TRUE, include=FALSE}
require(tidyverse); packageVersion("tidyverse")
require(phyloseq); packageVersion("phyloseq")
```

# Fin most abundant taxa across sites:

```{r, message = FALSE, warning = FALSE}
biobakery_function <- ("~/Documents/GitHub/DivComAnalyses/R/phyloseq_biobakery_functions.R")
alpha <- ("~/Documents/GitHub/DivComAnalyses/R/phyloseq_alpha.R")
beta <- ("~/Documents/GitHub/DivComAnalyses/R/phyloseq_beta.R")
taxa <- ("~/Documents/GitHub/DivComAnalyses/R/phyloseq_taxa_tests.R")
norm <- ("~/Documents/GitHub/DivComAnalyses/R/phyloseq_normalisation.R")
heat <- ("~/Documents/GitHub/DivComAnalyses/R/phyloseq_heatmap.R")
```

```{r, message = FALSE, results= "hide", warning = FALSE, include = TRUE}
sapply(c(biobakery_function, alpha, beta, taxa, norm, heat), source, chdir = TRUE)
```

```{r, message = FALSE, results=FALSE}
input <- list(metpahlan_ps = "~/Documents/GitHub/oral-microbiome/data/processed/metaphlan_ps.RDS",
              motus_ps = "~/Documents/GitHub/oral-microbiome/data/processed/motus/physeq.RDS",
              pathway_ps = "~/Documents/GitHub/oral-microbiome/data/processed/humann_pw_ps.RDS",
              kegg_ps = "~/Documents/GitHub/oral-microbiome/data/processed/humann_kegg_ps.RDS",
              l4c_ps = "~/Documents/GitHub/oral-microbiome/data/processed/humann_l4c_ps.RDS",
              go_ps = "~/Documents/GitHub/oral-microbiome/data/processed/humann_go_ps.RDS",
              infogo_ps = "~/Documents/GitHub/oral-microbiome/data/processed/humann_infogo_ps.RDS",
              metac_ps = "~/Documents/GitHub/oral-microbiome/data/processed/humann_metac_ps.RDS",
              pfam_ps = "~/Documents/GitHub/oral-microbiome/data/processed/humann_pfam_ps.RDS")

```

Data loading:

```{r, warning =FALSE, message = FALSE}
purrr::map(input, 
           readRDS) -> objects
```

Investigate most abundant taxa over sites:

```{r, message = FALSE, warning = FALSE}
sample_data(objects$metpahlan_ps$physeq)$Site_Health <- paste0(sample_data(objects$metpahlan_ps$physeq)$Oral_Site,
                                                               "_",
                                                               sample_data(objects$metpahlan_ps$physeq)$Health_status)
```

```{r, message = FALSE, warning = FALSE}

physeq_most_abundant <- function(physeq,
                                 group_var,
                                 ntax = 20,
                                 tax_level = "Species"){
  
  require(tidyverse); require(phyloseq)
  
  taxa_top_all = NULL  
  
  for(tp in physeq %>% 
      get_variable(group_var) %>%
      unique()){
    # print(tp)
    prune_samples(get_variable(physeq, group_var) == tp,
                  physeq) %>%
      fantaxtic::get_top_taxa(n = ntax, relative = FALSE, discard_other = TRUE) -> tmp2
    
    as(tax_table(tmp2), "matrix") %>%
      data.frame() %>%
      # dplyr::filter(grepl("ose",Gene_name)) %>%
      pull(tax_level) -> spc
    
    c(spc, taxa_top_all) %>%
      # discard(is.na)  %>%
      unique() %>%
      sort() -> taxa_top_all
    
    return(taxa_top_all)
  }
}


physeq_most_abundant(physeq = objects$metpahlan_ps$physeq,
                     group_var = "Site_Health",
                     ntax = 10) -> most_abundant

most_abundant
```

Heatmap those:
```{r, message = FALSE, warning = FALSE}
objects$metpahlan_ps$physeq %>%
  microbiome::transform('compositional') %>%
  phyloseq::subset_taxa(Species %in% most_abundant) %>%
  phyloseq_ampvis_heatmap(transform = "identity",
                          group_by = "Subject",#treatment
                          facet_by = c("Oral_Site", "Health_status", "Subject"),
                          tax_show = 100,
                          # order_x_by = "cluster",
                          min_abundance = 0,
                          tax_aggregate = "Species",
                          tax_add = NULL,
                          plot_na = FALSE,
                          color_vector = c("white", "red"),
                          plot_colorscale = "sqrt",
                          plot_legendbreaks = c(1, 10, 20)
  ) -> p

p + facet_grid(. ~ Oral_Site + Health_status, scales = "free", space = "free") + 
  scale_fill_viridis_c(breaks = c(0,  0.01, 1, 10, 50, 75, 100), 
                       labels = c(0,  0.01, 1, 10, 50, 75, 100), 
                       trans = scales::pseudo_log_trans(sigma = 0.01))  + 
  theme_classic() + 
      theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 6)) + 
      theme(axis.text.y = element_text(angle = 0,  size = 8))
```

```{r, message = FALSE, warning = FALSE}
RNA %>%
  filter(!is.na(Genus)) %>%
  top_n(10)
```

Let's combine the data from metagenomes and metatranscriptomes:
```{r, message = FALSE, warning = FALSE}
left_join(DNA,RNA, by= c("Full"="Full"), suffix = c("_DNA", "_RNA"), keep = FALSE) %>% #-> DNA_RNA
  mutate_if(is.numeric, ~replace(., is.na(.), 0)) -> DNA_RNA
# rows in x with no match in y will have NA values in the new columns
dim(DNA); dim(RNA); dim(DNA_RNA)
DNA_RNA %>%
  colnames()
```

```{r, message = FALSE, warning = FALSE}
# DNA_RNA %>%
#   mutate(Genus_DNA = as.factor(Genus_DNA)) %>%
#   mutate(Species_DNA = as.factor(Species_DNA)) %>%
#   mutate(Genus_DNA, fct_explicit_na(Genus_DNA, na_level = "unclassified")) %>%
#   mutate(Species_DNA, fct_explicit_na(Species_DNA, na_level = "unclassified"))
DNA_RNA %>%
  mutate(Species_DNA = if_else(is.na(Species_DNA), "unclassified", Species_DNA)) -> DNA_RNA

```
Unsurprisingly, more genes identified in metagenomes/ metatranscriptomes.

# Combine everything into a phyloseq object:

Clean a bit and generate a count and tax objects to be able to use phyloseq:

```{r, message = FALSE, warning = FALSE}
DNA_RNA %>%
  select(c("Full", "Gene_DNA","organism_DNA", "Genus_DNA", "Species_DNA")) %>%
  # separate(col=Description_DNA,
  #      into=c("Description","EC_number"),
  #      sep = "_EC_", remove = TRUE) %>%
  rename(Gene = Gene_DNA ,organism = organism_DNA, Genus = Genus_DNA , Species = Species_DNA) -> tax

# tax$GeneName <- tax$Gene_name
# # str_split_fixed(tax$Gene_name, "_EC_", n = Inf)
# tax$GeneName <- gsub("[^[:alnum:] ]", "_",tax$GeneName)
# tax$GeneName <- gsub(" ", "_",tax$GeneName)

DNA_RNA %>%
  select_if(is.numeric) -> count
# colnames(df) <- gsub("[^[:alnum:] ]", "",colnames(df))

dim(tax); dim(count)
```


```{r, message = FALSE, warning = FALSE}
physeq <- merge_phyloseq(otu_table(count %>% as.matrix(), 
                                   taxa_are_rows = TRUE),
                         tax_table(tax %>% as.matrix()
                         ),
                         sample_data(read.csv("/Users/fconstan/Projects/Oral/data/metadata_all_DNA_RNA.csv", header = TRUE ,
                                              row.names = 1 ,check.names = TRUE,
                                              na.strings=c("NA","NaN", ""),dec = ".", sep= ";"))
)


physeq
```
Seems to work !

Adjust metadata:
```{r, message = FALSE, warning = FALSE}
sample_data(physeq)$SampleID = rownames(sample_data(physeq))

sample_data(physeq)$SiteHealth = paste0(sample_data(physeq)$Oral_Site,"_",sample_data(physeq)$Health_status)

sample_data(physeq)$Health_status = factor(sample_data(physeq)$Health_status,
                                           levels = c("Healthy control",
                                                      "Caries",
                                                      "Periodontitis"))



sample_data(physeq)$Oral_Site = factor(sample_data(physeq)$Oral_Site,
                                       levels = c("SUBGINGIVAL_PLAQUE",
                                                  "TONGUE_BIOFILM",
                                                  "SALIVA"))

sample_data(physeq)$ReadNorm = ifelse(sample_data(physeq)$Type == "DNA", 
                                      sample_data(physeq)$DNA_non_human,
                                      sample_data(physeq)$RNA_non_human_mRNA
)


physeq %>%
  sample_data() %>%
  data.frame() %>%
  select(SampleID, everything()) -> meta

# physeq %>%
#   saveRDS(paste0("Oral_physeq_",table,".rds"))
merge_phyloseq(
  otu_table(physeq),
  sample_data(meta),
  tax_table(physeq)
)  -> physeq

physeq

# physeq %>%
#     subset_taxa(is.na(Genus)) %>%
#     subset_taxa(!(grepl("^UN", Full))) -> physeq # to get rid of taxonomic stratification

```

get unstratified output:
```{r message=FALSE, warnings = FALSE}
physeq %>%
  tax_table() %>%
  head()
```

```{r message=FALSE, warnings = FALSE}
physeq %>%
  tax_table() %>%
  tail()
```

create a reads counts phyloseq object (for alpha diversity - because it only takes integers...)
```{r message=FALSE, warnings = FALSE}
# https://bioconductor.org/packages/release/data/experiment/vignettes/curatedMetagenomicData/inst/doc/curatedMetagenomicData.html

sweep(physeq %>%
        otu_table() %>% 
        magrittr::divide_by(sample_sums(physeq)), 2, 
      sample_data(physeq)$ReadNorm , "*") %>%
  round() %>%
  as.data.frame() %>%
  # replace_na(list(0))
  select_if(~ !any(is.na(.))) -> counts

counts %>%
  as_tibble()
```

```{r message=FALSE, warnings = FALSE}
plot(counts %>%
       colSums(),
     sample_data(physeq)$ReadNorm)
```

```{r message=FALSE, warnings = FALSE}
merge_phyloseq(
  otu_table(counts, taxa_are_rows = TRUE),
  sample_data(physeq),
  tax_table(physeq)
)  -> phylo_count
```

```{r message=FALSE, warnings = FALSE}
phylo_count %>%
  taxa_sums() %>%
  sort(decreasing = TRUE) %>%
  head()

phylo_count %>%
  sample_sums() %>%
  sort(decreasing = TRUE)  %>%
  plot()

phylo_count %>%
  transform_sample_counts(function(x) x / sum(x) * 1000000) %>%
  sample_sums() %>%
  sort(decreasing = TRUE) %>%
  plot()


phylo_count %>%
  rarefy_even_depth()
```


```{r message=FALSE, warnings = FALSE}
physeq %>%
  taxa_sums() %>%
  sort(decreasing = TRUE) %>%
  head()

physeq %>%
  sample_sums() %>%
  sort(decreasing = TRUE)  %>%
  plot()

physeq %>%
  transform_sample_counts(function(x) x / sum(x) * 1000000) %>%
  sample_sums() %>%
  sort(decreasing = TRUE) %>%
  plot()

```


```{r message=FALSE, warnings = FALSE}
table="pathway"
prefix="pathway"

out <- list("phylo_count" = phylo_count, "physeq" = physeq)

assign(paste0(table, "_", prefix,"_physeqs"),
       out)

saveRDS(paste0(table, "_", prefix,"_physeqs") %>% get(),
        paste0("physeq_", table, "_", prefix,".RDS"))

save(list = c("phylo_count", "physeq"),
     file = paste0("physeq_", table, "_", prefix,".RData"))

# readRDS("physeq_pathway_pathway.RDS")
# load("physeq_pathway_pathway.RData")
```

# From: <https://github.com/karkman/MetagenomeCourse2019/tree/master/R_for_Humann2>

```{r message=FALSE, warnings = FALSE}

DNA_path %>%
  read_tsv() -> humann2_pathways

```

## Re-format the data frame from wide to long format
```{r message=FALSE, warnings = FALSE}
humann2_pathways %>% 
  # head(1000) %>%
  #Rename  `# Pathway` as pathway
  rename(pathway = `# Pathway`) %>%
  rename_at(vars(ends_with("_R1R2_cat_fastq_Abundance-RELAB")),
            funs(str_replace(., "_R1R2_cat_fastq_Abundance-RELAB", "_DNA"))) %>%
  #Gather cmp by pathway and sampleID
  # gather(sampleID, cpm, -pathway) %>% 
  # pivot_longer(cols = c("DNA", "RNA"),
  #             values_to = "Abundance",
  #             names_to = 'Type') %>%
  pivot_longer(cols = `4012_SALIVA_DNA`:`MP9_SUB_DNA`,
               values_to = "cpm",
               names_to = "SampleID") %>% 
  #Separate by sampleID and drop any extra values without warning
  # separate(sampleID, "sampleID", sep = "_", extra = "drop") %>% 
  #Separate pathways from organisms using |
  separate(pathway, c("pathway", "organism"), sep = "\\|", fill = "right") %>%
  separate(organism, c("Genus", "Species"), sep = "\\.", fill = "right", remove = FALSE) %>%
  mutate(organism = str_replace(organism, ".s__", "_")) %>% 
  mutate(organism = str_replace(organism, "g__|", "")) %>% 
  mutate(Genus = str_replace(Genus, "g__", "")) %>% 
  mutate(Genus = str_replace(Species, "s__", "")) -> humann2_pathways_long

```

## Continue processing the humann2 output:

```{r message=FALSE, warnings = FALSE}
# generate pathway table with no organism stratifications
humann2_pathways_long %>%
  filter(is.na(organism)) %>%
  select(-organism) %>%
  filter(!(grepl("^UN", pathway))) -> humann2_pathways_no_stratifications_long

humann2_pathways_no_stratifications_long
```

```{r message=FALSE, warnings = FALSE}
humann2_pathways_no_stratifications_long %>% group_by(SampleID) %>% summarise(Total = sum(cpm)) %>% arrange(-Total)
```

```{r message=FALSE, warnings = FALSE}

# Compute pathway alpha divertities per sample
humann2_pathways_no_stratifications_long %>%
  group_by(SampleID) %>%
  summarise(shannons_div = vegan::diversity(cpm),
            invsimpson_div = vegan::diversity(cpm, index = "invsimpson"),
            richness = sum(cpm>0)) %>%
  arrange(richness)
```

```{r message=FALSE, warnings = FALSE}
# Work with organism level stratifications
humann2_pathways_long %>%
  filter(!(is.na(organism))) %>%
  filter(!(grepl("^UN", pathway))) -> humann2_pathway_stratifications_long

humann2_pathway_stratifications_long %>% group_by(SampleID) %>% summarise(Total = sum(cpm)) %>% arrange(-Total)
```

```{r message=FALSE, warnings = FALSE}
# number of organisms per pathway
humann2_pathway_stratifications_long %>%
  group_by(pathway, SampleID) %>%
  summarise(num_organisms = length(unique(organism))) -> humann2_organisms_per_pathway 

# average contributional alpha diverity (Gini simpson diversity) per pathway
humann2_pathway_stratifications_long %>%
  filter(cpm > 0) %>%
  group_by(pathway, SampleID) %>%
  summarise(shannons_div = vegan::diversity(cpm),
            invsimpson_div = vegan::diversity(cpm, index = "invsimpson"),
            richness = sum(cpm>0)) %>%
  group_by(pathway) %>%
  summarise(mean_alpha_div = mean(shannons_div),
            median_alpha_dv = median(shannons_div)) %>%
  arrange(-mean_alpha_div) -> humann2_pathway_alpha_div
head(humann2_pathway_alpha_div)
```

```{r message=FALSE, warnings = FALSE}
# add number of samples where pathway present
humann2_pathways_no_stratifications_long %>%
  filter(cpm > 0) %>%
  group_by(pathway) %>%
  summarise(n_samples = n()) %>%
  left_join(humann2_pathway_alpha_div) %>%
  left_join(humann2_organisms_per_pathway) -> humann2_pathway_alpha_div_with_n_samples

humann2_pathway_alpha_div_with_n_samples
```
```{r message=FALSE, warnings = FALSE}
humann2_pathway_alpha_div_with_n_samples %>%
  filter(n_samples > 10) %>%
  arrange(-mean_alpha_div) %>%
  print(n = 20)
```

```{r message=FALSE, warnings = FALSE}
sessionInfo()
```
